### PR TITLE
Update snctl to v0.14.0

### PR DESCRIPTION
**Pull request by EronWright**
Automated changes by Release workflow in streamnative/cloud-cli repository.

### CODE REVIEW

**Review by jung-kim — APPROVED** [review]

**Comment by EronWright** [comment]
Feel free to merge as soon as is approved.

**Review by nlu90 — APPROVED** [review]

File: Formula/snctl.rb [hunk 1]
```diff
@@ -5,21 +5,21 @@
 class Snctl < Formula
   desc "StreamNative Cloud CLI (snctl)"
   homepage "https://streamnative.io/"
-  version "0.13.3"
+  version "0.14.0"
   license "Apache-2.0"
 
   on_macos do
-    if Hardware::CPU.intel?
```
Comment (heesung-sn) [inline Formula/snctl.rb:12]: I guess we don't have mac && intel cases anymore?

Comment (EronWright) [reply Formula/snctl.rb:12]: This was published using goreleaser, and I do think that mac/intel and mac/arm are supported.  I don't know the details of the generated brew script.
https://github.com/streamnative/cloud-cli/releases/tag/v0.14.0

**Comment by EronWright** [comment]
Verified:
```
$ brew upgrade snctl
==> Upgrading 1 outdated package:
streamnative/streamnative/snctl 0.13.3 -> 0.14.0
==> Fetching streamnative/streamnative/snctl
==> Downloading https://storage.googleapis.com/downloads.streamnative.cloud/snctl/v0.14.0/snctl_0.14.0_darwin_amd64.tar.gz
######################################################################## 100.0%
==> Upgrading streamnative/streamnative/snctl
  0.13.3 -> 0.14.0

```

**Pull request merged**